### ISSUE TITLE
Identify methods used single time

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#948]
Imagine the following scenario: you create an `class Child extens Parent`, and a method on `Parent` that is used exclusively on `Child` class. Seems more logical you move this method to `Child` instead of keep it on `Parent`.

```php
class Parent {
    public function doSomething() {}
}

class Child extends Parent {
    // The unique code that calls the doSomething() from parent.
    public function callDoSomething() { $this->doSomething(); }
}
```

I think that it is a valid idea, but have some problems, for instance: IDE could not identify correctly the usage of `doSomething()` in some cases:

```php
$doSomething = 'doSomething';
call_user_func([ new Parent, $doSomething ]);
```

But in case above, it seems a bad pratice. I think that with good pratices, it could be a valid inspection to avoid "overcharge" parent with methods used only by the child.

**Comment by kalessil** [comment]
@rentalhost : does it make more sense to extend senseless proxy function here, so it encourages people to use the parent method directly?

**Comment by rentalhost** [comment]
I don't understand what you mean here. I don't know too much the proxy pattern, so is complicated to me haha

**Comment by rentalhost** [comment]
I do not know... Perhaps the biggest problem here is a possible rarity of occasions for this to happen. I found an interesting idea once I was refactoring a code and realized that I defined a method in parent that was used only by one of its children. That is, the ideal was that this method existed in the child, and not in the parent.
